### PR TITLE
prrte uses libev data structues, not libevent

### PR DESCRIPTION
**Pull request by roblatham00**
the 'libev' package provides data structures prrte needs such as
`ev_async`.  libevent does not offer such an interface.  Was it a mistake to depend on `libevent` instead of `libev`?  

Without this change prrte fails to build, giving errors:

```==> Error: ProcessError: Command exited with status 2:
    'make' '-j4'

4 errors found in build log:
     1645      CC       runtime/opal_init.lo
     1646      CC       runtime/opal_params.lo
     1647      CC       runtime/opal_info_support.lo
     1648      CC       runtime/opal_progress_threads.lo
     1649      CC       threads/condition.lo
     1650      CC       threads/mutex.lo
  >> 1651    runtime/opal_progress_threads.c:50:5: error: unknown type name 'ev_async'
     1652       50 |     ev_async async;
     1653          |     ^~~~~~~~
  >> 1654    runtime/opal_progress_threads.c:116:37: error: unknown type name 'EV_P_'
     1655      116 | static void opal_libev_ev_async_cb (EV_P_ ev_async *w, int revents)
     1656          |                                     ^~~~~  
     1657    runtime/opal_progress_threads.c: In function 'opal_event_add':
     1658    runtime/opal_progress_threads.c:150:9: warning: implicit declaration of function 'ev_async_send' [-Wimplicit-function-declaration]
     1659      150 |         ev_async_send ((struct ev_loop *)trk->ev_base, &trk->async);
     1660          |         ^~~~~~~~~~~~~
     1661    runtime/opal_progress_threads.c: In function 'opal_progress_thread_init':
     1662    runtime/opal_progress_threads.c:320:5: warning: implicit declaration of function 'ev_async_init'; did you mean 'evtag_init'? [-Wimplicit-function-declaration]
     1663      320 |     ev_async_init (&trk->async, opal_libev_ev_async_cb);
     1664          |     ^~~~~~~~~~~~~
     1665          |     evtag_init
  >> 1666    runtime/opal_progress_threads.c:320:33: error: 'opal_libev_ev_async_cb' undeclared (first use in this function)
     1667      320 |     ev_async_init (&trk->async, opal_libev_ev_async_cb);
     1668          |                                 ^~~~~~~~~~~~~~~~~~~~~~
     1669    runtime/opal_progress_threads.c:320:33: note: each undeclared identifier is reported only once for each function it appears in
     1670    runtime/opal_progress_threads.c:321:5: warning: implicit declaration of function 'ev_async_start' [-Wimplicit-function-declaration]
     1671      321 |     ev_async_start((struct ev_loop *)trk->ev_base, &trk->async);
     1672          |     ^~~~~~~~~~~~~~
     1673    runtime/opal_progress_threads.c: In function 'opal_progress_thread_attach':
  >> 1674    runtime/opal_progress_threads.c:382:33: error: 'opal_libev_ev_async_cb' undeclared (first use in this function)
     1675      382 |     ev_async_init (&trk->async, opal_libev_ev_async_cb);
     1676          |                                 ^~~~~~~~~~~~~~~~~~~~~~
     1677    make[2]: *** [Makefile:1448: runtime/opal_progress_threads.lo] Error 1
     1678    make[2]: *** Waiting for unfinished jobs....
     1679    make[2]: Leaving directory '/tmp/robl/spack-stage/spack-stage-prrte-1.0.0-wvg7qdwqtcsxyuqq7jpqqg45dtngh3xl/spack-src/opal'
     1680    make[1]: *** [Makefile:1520: all-recursive] Error 1
```

### CODE REVIEW

**Comment by adamjstewart** [comment]
@dorier

**Comment by mdorier** [comment]
I did the initial commit of the prrte package but I am not a developer of prrte. I might have made a mistake making it depend on libevent, but the best would be to confirm with a prrte developer.

**Comment by roblatham00** [comment]
yeah, let's hold off on merging this until I can test this more thoroughly.    I'm able to build, but cannot actually use prrte with this change.  Going to investigate some more.

**Comment by rhc54** [comment]
I'm not entirely sure of the scenario here. Are you trying to build PRRTE v1.0? Or are you trying to build PRRTE master (soon to be v2.0)?

PRRTE should support either libevent or libev, but will only use the latter (libev) if you explicitly request it with the `--with-libev` configure flag. My guess is that somewhere in your build logic you included the `--with-libev` flag, but actually provided the libevent library?

I haven't checked PRRTE v1.0 in a very long time. However, we regularly build/test PRRTE master with libevent as that is our default event library. I honestly haven't checked it with libev in some time, but I understand that at least one user regularly checks that dependency - I can double-check with them to ensure it still works.

**Comment by adamjstewart** [comment]
@rhc54 Thanks, that's very helpful! Would you be interested in being listed as an official "maintainer" for this Spack package? You don't have to be a Spack expert, basically it just gives us someone to ping when we have questions about a PR or someone encounters build issues. If so, we can add this to the package:
```python
maintainers = ['rhc54']
```
We can list other GitHub accounts as well.

**Comment by rhc54** [comment]
Sure - happy to do so.

FWIW: I just committed changes to both the PRRTE and PMIx configure logic to protect against such an error in the future. So if someone accidentally tries to configure `--with-libev` and passes in the path to `libevent`, we will error out during configure.

HTH

**Comment by roblatham00** [comment]
@rhc54 correctly spelled out the problelm.  I updated my pull request so that now PRRTE requests the `libevent` package and passed the `libevent` library to it.    PMIx is using libevent and it seems from my experience that PRRTE should use the same library.

**Comment by rhc54** [comment]
> PRRTE should use the same library.

Correct. Sadly, libevent and libev have common symbols, so if PMIx is using one and PRRTE the other, the two libraries will both be pulled into memory and you'll get symbol confusion.

**Comment by adamjstewart** [comment]
@roblatham00 can you add @rhc54 as a maintainer for this package?

**Comment by roblatham00** [comment]
Sure thing. Just pushed a commit doing so.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**
